sql/catalyst/src/main/java/org/apache/spark/sql/connector/util/V2ExpressionSQLBuilder.java

```diff
@@ -393,10 +394,11 @@ private String[] expressionsToStringArray(Expression[] expressions) {
     return result;
   }
 
-  private List<String> expressionsToStringList(Expression[] expressions) {
-    List<String> list = new ArrayList<>(expressions.length);
-    for (Expression expression : expressions) {
-      list.add(build(expression));
+  private List<String> expressionsToStringList(Expression[] expressions, int offset, int length) {
+    List<String> list = new ArrayList<>(length);
+    while (offset < offset + length) {
+      list.add(build(expressions[offset]));
+      offset++;
```

Comment:
This looks wrong ...
Probably you meant something like :
```
final int till = Math.min(offset + length, expressions.length);
while (offset < till) {
  list.add(build(expressions[offset]));
  offset ++;
}

```

Reply:
Oh, my God... It's a shame ...... :(